### ISSUE TITLE
merging fails in some cases:

### ISSUE THREAD

**Issue by nathandunn** [GMOD/Apollo#850]
Found in #842, but I was able to reproduce this with 2.0.1, but only was able to do it for the exact gene pairing in Group1.33 (as opposed to 1.10). 

Bug 3: add A1 B1, modify B1 (delete exon A1 B2), modify A1 (Set readthrough stop codon = A2 B2), merge (expected AB, but fails) 

Bug 3 described here:
[issues_on_pull_request_842.pdf](https://github.com/GMOD/Apollo/files/116081/issues_on_pull_request_842.pdf)

Stacktrace of failure in 2.0.1:
```
2016-02-03 09:35:42,104 [clientInboundChannel-4] ERROR springwebsocket.GrailsSimpAnnotationMethodMessageHandler  - Unhandled exception
java.lang.reflect.InvocationTargetException
	at org.bbop.apollo.AnnotationEditorController$__do_annotationEditor_closure3_closure4_closure5.doCall(AnnotationEditorController.groovy:1075)
	at org.bbop.apollo.AnnotationEditorController$__do_annotationEditor_closure3_closure4.doCall(AnnotationEditorController.groovy:1074)
	at org.bbop.apollo.AnnotationEditorController$__do_annotationEditor_closure3.doCall(AnnotationEditorController.groovy:1070)
	at groovyx.gpars.group.PGroup$3.call(PGroup.java:289)
	at groovyx.gpars.group.PGroup$4.run(PGroup.java:313)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
Caused by: org.springframework.dao.InvalidDataAccessApiUsageException: deleted object would be re-saved by cascade (remove deleted object from associations): [org.bbop.apollo.FeatureRelationship#205086]; nested exception is org.hibernate.ObjectDeletedException: deleted object would be re-saved by cascade (remove deleted object from associations): [org.bbop.apollo.FeatureRelationship#205086]
	at org.grails.datastore.gorm.GormStaticApi$_methodMissing_closure2.doCall(GormStaticApi.groovy:102)
	at org.bbop.apollo.FeatureRelationshipService.$tt__removeFeatureRelationship(FeatureRelationshipService.groovy:168)
	at org.bbop.apollo.TranscriptService.$tt__deleteTranscript(TranscriptService.groovy:110)
	at org.bbop.apollo.TranscriptService.$tt__mergeTranscripts(TranscriptService.groovy:461)
	at org.bbop.apollo.RequestHandlingService.$tt__mergeTranscripts(RequestHandlingService.groovy:2120)
	... 8 more
Caused by: org.hibernate.ObjectDeletedException: deleted object would be re-saved by cascade (remove deleted object from associations): [org.bbop.apollo.FeatureRelationship#205086]
	... 13 more
```

Failure in 2.0.2-SNAPSHOT:

<img width="842" alt="screen shot 2016-02-03 at 9 47 29 am" src="https://cloud.githubusercontent.com/assets/751274/12791276/3c5e30b4-ca5b-11e5-8f0e-44237bbd20df.png">

**Comment by nathandunn** [comment]
@deepakunni3  Can you take a look at this.  I think that this is separate from the history bug.

**Comment by deepakunni3** [comment]
@nathandunn This would most likely be because its failing to remove the featureProperty 'set read through stop codon' at the time of merge (a guess based on the stack trace) or its trying to remove the property after the transcript is deleted (post merge).

I'll look into it.

**Comment by nathandunn** [comment]
I  hadn't thought about the Feature Properties, but that makes sense.  Probably just need to make sure those get explicitly removed from the relationship in the same way that relationships get removed.

**Comment by monicacecilia** [comment]
@nathandunn 
Fixed: 
- [x] Bug 1: Fixed.
- [x] Bug 2 It is 'technically' fixed-- but with caveats. First, by the time I tried to test, History had already been lost. As reported in #792 (see image below). And It went back to state B1, but resulting transcript was not visible at first, so had to refresh. :disappointed:  ![screen shot 2016-03-03 at 7 01 40 pm](https://cloud.githubusercontent.com/assets/1714621/13516806/70eb6fd0-e172-11e5-9881-8734398b48ad.png) 
- [x] Bug 3: Great. Does not give an error anymore.  
- [x] Bug 4: Histories appropriately correspond to respective transcripts.
